torch/_inductor/codegen/wrapper.py

```diff
@@ -911,28 +920,63 @@ def write_async_compile_wait(self) -> None:
             """
         )
 
+    def write_args(self, input_names: List[str]):
+        lhs = ", ".join(input_names)
+        if len(input_names) == 1:
+            lhs += ","
+        self.prefix.writeline(f"{lhs} = args")
+        self.prefix.writeline("args.clear()")
+
     def write_prefix(self) -> None:
         assert self.launcher_fn_name is not None
         self.write_async_compile_wait()
-        self.prefix.splice(
-            f"""
-            def {self.launcher_fn_name}(args):
-            """
-        )
-        with self.prefix.indent():
+        if (
+            config.graph_partition
+            and not V.graph.cpp_wrapper
+            and not V.graph.aot_mode
+            and not hasattr(self, "subgraph_name")
```

Comment:
hmm, i wonder if theres a better way of structuring code then the `hasattr(self, "subgraph_name"`. maybe checking the type of self, or defining a method with different impls.